codeclimate-wrapper

```diff
@@ -18,32 +18,52 @@ EOF
   exit 1
 }
 
+docker_for_mac() {
+  test -S /var/tmp/docker.sock > /dev/null 2>&1
+}
+
+socket_exists() {
+  test -S $socket_path > /dev/null 2>&1
+}
+
+socket_path() {
```

Comment:
Ah good note.

What about this:

1. If the current version is symlinking to `/var/run/docker.sock` we should just use that and not bother an alternate `$socket_path`
1. If the documententation directs uses to unset `$DOCKER_`, we should rely on that and not worry about short-circuiting before checking for docker-machine
1. This only leaves the potential for a user having `boot2docker` on `$PATH`, but actually intending to use Docker Machine -- I think it's OK not to handle this edge-case

If those three things are true / appropriate -- do we need any changes at all?